### DISCUSSION TITLE
Migrating from docker to non-docker: Is Redis database re-generatable or mergeable?

### DISCUSSION THREAD

**Discussion by lo48576** [bookwyrm-social/bookwyrm#3464] [Q&A] [answered]
I'm migrating from docker-compose deployment to non-docker, and it almost succeeded... without Redis part.

Current docker-compose.yml on `production` branch [is using two Redis 7.2.1 servers for `redis_activity` and `redis_broker`](https://github.com/bookwyrm-social/bookwyrm/blob/f7958bc0f309cd80927b1c5b8c2d429f44261743/docker-compose.yml#L66), but the latest package on Debian 12 (bookworm) [is Redis 7.0.15](https://packages.debian.org/ja/bookworm/redis-server) and [the official documentation share the `localhost` for two purposes](https://docs.joinbookwyrm.com/install-prod-dockerless.html).

So I googled a bit but cannot find the ways to:

* moving the Redis data from newer (7.2.1) DB to older (7.0.15) DB, and
* merging two databases (`redis_activity` and `redis_broker` in docker compose) into one (redis.service on Debian).

So my question is, how this migration can be done?
Is there any ways to successfully migrate Redis data?
Or the data on Redis can be re-generated from the data on Postgresql?

**Comment by lo48576** [comment]
[Updating Your Instance](https://docs.joinbookwyrm.com/updating.html)
> Feeds for each user are stored in Redis. To re-populate a stream, use the management command:

Oops I overlooked this, but is this enough for filling empty Redis database?
I assume so, but I'm not exactly sure what the two Redis servers do and whether it is safe to resetting them.

**Reply by mouse-reeve** [reply, marked as answer]
Redis activity can be completely regenerated from nothing using the script you mention. Redis broker however stores data for celery, and can’t be rebuilt. It handles executing longer running background scripts and communicating with external instances. If there are no pending celery tasks, deleting the data won’t cause any disruption to service, but that’s not quite the same thing.

**Reply by lo48576** [reply]
Thank you.

> It handles executing longer running background scripts and communicating with external instances. If there are no pending celery tasks,

Would it be (relatively) safe to ignore some "longer running background scripts"?
My instance is a personal single-user almost-no-followers server, so one-time inconsistency on federation is acceptable for me.
If the bad effect of "disruption" does not persist locally, I'm OK to regard the migration as completed (without checking whether some tasks had been pending).

**Reply by lo48576** [reply]
Closing as no problem since my server seems to work properly (at least as far as I observe), though I didn't carefully investigated what can happen when Redis DB is cleared.
